pkgs/build-support/go/module.nix

```diff
@@ -175,10 +180,14 @@ in
         (lib.optional (!finalAttrs.proxyVendor) "-mod=vendor")
       ++ lib.warnIf (builtins.elem "-trimpath" GOFLAGS) "`-trimpath` is added by default to GOFLAGS by buildGoModule when allowGoReference isn't set to true"
         (lib.optional (!allowGoReference) "-trimpath");
-    inherit CGO_ENABLED enableParallelBuilding GO111MODULE GOTOOLCHAIN;
+    inherit enableParallelBuilding GO111MODULE GOTOOLCHAIN;
+
+    CGO_ENABLED = linuxGnuPie || CGO_ENABLED;
```

Comment:
This doesn't evaluate, bool and int.

Reply:
..or string. CGO_ENABLED = "0" is also common